src/imu_control_parameters.cpp

```diff
@@ -436,11 +457,30 @@ void ImuControlParameters::setParametersInt32()
             "error on set parameter %s with %d value; current value remained %d", ckey,
             requestedValue, m_int32_current_params[key]);
         } else {
-          RCLCPP_INFO(
-            rclcpp::get_logger("rclcpp_imucontrolparameter"),
-            "successfully set parameter %s: old value = %d new value = %d", ckey,
-            m_int32_current_params[key], requestedValue);
-          m_int32_current_params[key] = requestedValue;
+          // readback value from hardware
+          if (!(m_iio_wrapper.*(m_func_map_get_int32_params[key]))(i32DriverParam)) {
```

Comment:
i32DriverParam shouldn't be passeed by pointer

Reply:
Please see the following:
```
  /*! Declare function with int32_t& parameter and return bool */
  typedef bool (IIOWrapper::*GetInt32Params)(int32_t &);

  /*! Declare a map with GetInt32Params functions */
  typedef std::map<std::string, GetInt32Params> GetInt32ParamsMapType;

  /*! Declare a variable of GetInt32ParamsMapType */
  GetInt32ParamsMapType m_func_map_get_int32_params;
```
And 
```
  /*! Declare function with double parameter and return bool */
  typedef bool (IIOWrapper::*GetDoubleParams)(double *);

  /*! Declare a map with GetDoubleParams functions */
  typedef std::map<std::string, GetDoubleParams> GetDoubleParamsMapType;

  /*! Declare a variable of GetDoubleParamsMapType */
  GetDoubleParamsMapType m_func_map_get_double_params;
```


In this case it is an address. only in the case of m_func_map_get_double_params a pointer is expected. 
We plan to perform a refactoring which will use pointers everywhere to avoid confusion...but this would be done in a different PR.

Reply:
Okay, i see. thanks!